### PR TITLE
chore(build): clean up dependencies and convert all tests to jest

### LINKED ISSUE #106: Build is broken for lerna bootstrap
Lerna bootstrap throws errors and the root level package has missing dependencies for the clean script.

### PR DESCRIPTION
**Pull request by janechu**
Closes #106

### CODE REVIEW

File: build/clean.js [hunk 1]
```diff
@@ -15,10 +15,20 @@ const paths = argv._;
...
-    const removePath = path.resolve(process.cwd(), cleanPath)
-    rimraf(removePath, () => {
-        console.log(removePath, "cleaned");
-    });
+    if (fs.existsSync(path)) {
+        fs.readdirSync(path).forEach(function(file, index){
+            var currentPath = `${path}/${file}`;
+            
+            if (fs.lstatSync(currentPath).isDirectory()) { // recurse
+                deleteFolderRecursive(currentPath);
```
Comment (nicholasrice) [inline build/clean.js]: where is this coming from? Maybe just leave rimraf if you're gonna have to write all of this.

File: packages/fast-permutator/__tests__/test-utils/index.js [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+
+/**
+ * Exported method for getting shallow refs
+ */
+
+describe('Reference IDs', function() {
+    let possibleRefs;
+
+    beforeAll((done) => { // done
+        let schemaData = require('../schemas/primitive/ref.schema.json');
```
Comment (nicholasrice) [inline packages/fast-permutator/__tests__/test-utils/index.js]: does this really need to be async? It doesn't look like you're calling done based on any async process. Same for other `beforeAll` functions.

File: packages/fast-permutator/__tests__/test-utils/index.js [hunk 3]
```diff
@@ -0,0 +1,283 @@
+var get = require('lodash').get,
```
Comment (chrisdholt) [inline packages/fast-permutator/__tests__/test-utils/index.js]: are we avoiding const here for a reason? I believe that the issue Aaron was running into was due to the fact that TravisCI was running in node v 0.10 - he's update to use ^4.0 and const doesn't seem to be an issue anymore.

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+    "compilerOptions": {
+        "moduleResolution": "node",
+        "module": "ES6",
+        "target": "ES6",
+        "baseUrl": ".",
+        "declaration": true,
+        "outDir": "./dist"
+    },
+    "include": [
```
Comment (nicholasrice) [inline tsconfig.json]: minor nit, but should we use `src` so that it aligns with the rest of our repo?

**Pull request merged**